jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java

```diff
@@ -0,0 +1,145 @@
+
+import com.ibm.oti.vm.VM;
+import com.ibm.tools.attach.target.IPC;
+
+/**
+ * Common methods for the diagnostics tools
+ *
+ */
+public class DiagnosticUtils {
+
+	/**
+	 * Command strings for executeDiagnosticCommand()
+	 */
+
+	/**
+	 * Run System.gc();
+	 */
+	public static final String DIAGNOSTICS_GC_RUN = "GC.run"; //$NON-NLS-1$
+
+	/**
+	 * Get the heap object statistics.
+	 */
+	public static final String DIAGNOSTICS_GC_CLASS_HISTOGRAM = "GC.class_histogram"; //$NON-NLS-1$
+
+	/**
+	 * Key for the command sent to executeDiagnosticCommand()
+	 */
+	public static final String COMMAND_STRING = "command_string"; //$NON-NLS-1$
+
+	/**
+	 * Use this to separate arguments in a diagnostic command string.
+	 */
+	public static final String DIAGNOSTICS_OPTION_SEPARATOR = ","; //$NON-NLS-1$
+
+	/**
+	 * Report only live heap objects.
+	 */
+	public static final String LIVE_OPTION = "live"; //$NON-NLS-1$
+
+	private static final Map<String, Function<String, DiagnosticProperties>> commandTable;
+
+	/**
+	 * Create the command to run the heapHisto command
+	 * 
+	 * @param liveObjects add the option to run a GC before listing the objects
+	 * @return formatted string
+	 */
+	public static String makeHeapHistoCommand(boolean liveObjects) {
+		String cmd = DIAGNOSTICS_GC_CLASS_HISTOGRAM;
+		if (liveObjects) {
+			cmd = DIAGNOSTICS_GC_CLASS_HISTOGRAM+ DIAGNOSTICS_OPTION_SEPARATOR + LIVE_OPTION;
+		}
+		return cmd;
+	}
+
+	public static final String EMPTY_STRING = ""; //$NON-NLS-1$
+
+	private static native String getHeapClassStatisticsImpl();
+
+	/**
+	 * Run a diagnostic command and return the result in a properties file
+	 * 
+	 * @param diagnosticCommand String containing the command and options
+	 * @return command result or diagnostic information in case of error
+	 */
+	public static DiagnosticProperties executeDiagnosticCommand(String diagnosticCommand) {
+		IPC.logMessage("executeDiagnosticCommand: ", diagnosticCommand); //$NON-NLS-1$
+
+		DiagnosticProperties result;
+		String[] commandRoot = diagnosticCommand.split(DiagnosticUtils.DIAGNOSTICS_OPTION_SEPARATOR);
+		Function<String, DiagnosticProperties> cmd = commandTable.get(commandRoot[0]);
+		if (null == cmd) {
+			result = DiagnosticProperties.makeStatusProperties(true,
+					"Command " + diagnosticCommand + " not recognized"); //$NON-NLS-1$ //$NON-NLS-2$
+		} else {
+			result = cmd.apply(diagnosticCommand);
+			result.put(DiagnosticUtils.COMMAND_STRING, diagnosticCommand);
+		}
+		return result;
+	}
+
+	private static DiagnosticProperties getHeapStatistics(String diagnosticCommand) {
+		boolean doLive = Arrays.stream(diagnosticCommand.split(DiagnosticUtils.DIAGNOSTICS_OPTION_SEPARATOR))
+				.anyMatch(Predicate.isEqual(DiagnosticUtils.LIVE_OPTION));
+		if (doLive) {
+			runGC();
+		}
+		String result = getHeapClassStatisticsImpl();
+		String lineSeparator = System.lineSeparator();
+		final String unixLineSeparator = "\\n"; //$NON-NLS-1$
+		if (unixLineSeparator != lineSeparator) {
```

Comment:
Two problems here:
* the normal line separator is `\n`, not `\\n` (one character, not two)
* this should not assume the system line separator is interned (use `!equals`, instead of `!=`)

Reply:
Right.  Fixed.